### ISSUE TITLE
libwebsockets.org is down

### ISSUE THREAD

**Issue by mkohanim** [warmcat/libwebsockets#2841]
Please help. We have an outstanding build which requires the latest source files (4.3.2) but the site is down.

**Comment by Ximalas** [comment]
The FreeBSD ports collection ran into the same issue. They switched to this URL:

https://codeload.github.com/warmcat/libwebsockets/tar.gz/v4.3.2?dummy=/warmcat-libwebsockets-v4.3.2_GH0.tar.gz

**Comment by diizzyy** [comment]
It appears to be up now again

https://libwebsockets.org/git/libwebsockets/snapshot/libwebsockets-v4.3.2.tar.xz

**Comment by mkohanim** [comment]
Hi Daniel, yes, thank you. The website (/git) came backup yesterday and allowed us to proceed with our build.

With kind regards,

******************************
  Michel Kohanim
             CEO

  (p) 818.631.0333
  http://www.universal-devices.com<http://www.universal-devices.com/>
******************************

**Comment by mike-jumper** [comment]
It seems to be back down.

**Comment by lws-team** [comment]
Yeah... I know.  It's natural state isn't being up... I have to make arrangements, pay for it and keep it going.

**Comment by diizzyy** [comment]
Is it possible to upload release archives on GitHub as backup?

**Comment by lws-team** [comment]
I will bring my git back up today, but FYI github will give you a tarball of a tag or branch without needing me to spend time on it

curl -L  https://github.com/warmcat/libwebsockets/tarball/v4.3.2 -o dummy.tgz

**Comment by diizzyy** [comment]
Yes, but that's not static and unlikely to share the same checksum as the one from main site.

**Comment by lws-team** [comment]
If you want a tarball of a tag, you can have it today from github as shown.  If you have a higher grade of OCD than that, you can cache it or do whatever you like.

**Comment by diizzyy** [comment]
It's going to be an issue for pretty much any type of packaging (read pretty much any distro that tries to package it)

**Comment by lws-team** [comment]
What exactly do you think the problem is, that suddenly no build system can build a git tag, from github?

```
$ curl -L https://github.com/warmcat/libwebsockets/tarball/v4.3.2 -o dummy1.tgz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 14.8M    0 14.8M    0     0  8711k      0 --:--:--  0:00:01 --:--:-- 11.3M
$ curl -L https://github.com/warmcat/libwebsockets/tarball/v4.3.2 -o dummy2.tgz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 14.8M    0 14.8M    0     0  8176k      0 --:--:--  0:00:01 --:--:-- 11.0M
$ sha256sum dummy1.tgz 
e2006366aa1cbe57fcfa2afbe9617fcee1cfa3406ceb4c7b5fae78c98e087f2d  dummy1.tgz
$ sha256sum dummy2.tgz 
e2006366aa1cbe57fcfa2afbe9617fcee1cfa3406ceb4c7b5fae78c98e087f2d  dummy2.tgz
```

**Comment by diizzyy** [comment]
```
curl https://libwebsockets.org/git/libwebsockets/snapshot/libwebsockets-v4.3.2.tar.gz -o dummy.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 10.0M    0 10.0M    0     0  1794k      0 --:--:--  0:00:05 --:--:-- 2078k
```
sha256sum dummy.tar.gz
3f9a610dc94b5ab29a5a52ac389d4c2738b0f373eef455955654625888a90871  dummy.tar.gz

**Comment by lws-team** [comment]
This is only telling you about gitohashi's code for using libarchive not setting the file dates before compressing it.  It's not telling you about github or that there's any value in using github release things instead of the solution I gave you above, or that build systems can't just use git to fetch a tag, from github or whereever.

If you care about it enough to continue making me read about it, you're welcome to send a patch to force the date on the files before archiving them.

https://warmcat.com/git/gitohashi

**Comment by AlojzyB** [comment]
Is libwebsockets.org down again? Need urgently support and I was hope to find it there :(.

**Comment by diizzyy** [comment]
Yep, it's down again.
Can you perhaps upload static tarballs to the GitHub repo as backup?
See also https://github.blog/2023-02-21-update-on-the-future-stability-of-source-code-archives-and-hashes/
